lib/MusicBrainz/Server/Data/LinkType.pm

```diff
@@ -114,14 +114,27 @@ sub load
 
 sub get_tree
 {
-    my ($self, $type0, $type1) = @_;
+    my ($self, $type0, $type1, $get_deprecated_and_empty) = @_;
 
     my %id_to_obj;
     my @objs;
+    my $extra_condition = '';
+
+    unless ($get_deprecated_and_empty) {
+        $extra_condition = <<~'EOSQL';
+            AND NOT (
+                is_deprecated = TRUE
+                AND NOT EXISTS (
+                    SELECT 1 FROM link WHERE link.link_type = id
+                )
+            )
```

Comment:
The double negative is kinda hard to parse. How about:
```suggestion
            AND is_deprecated = FALSE
            AND EXISTS (SELECT 1 FROM link WHERE link.link_type = id)
```

Reply:
That's not what we want, though, it should show deprecated things as long as 1 exists in link, but I guess it could be simpler with those two conditions with an OR. Will do.